src/adr/articles/0012-aws-xray.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: layouts/adr.njk
+title: "AWS X-Ray"
+weight: 1
+date: 2022-03-21
+review_in: 12 months
+tags:  
+    - adr
+    - observability
+    - tools
+    - language
+areas_of_coverage: ["Digital Service"]
+status: "accepted"
+contributors: ["John Nolan"]
+adr_number: "0012"
+---
+
+## Context
+
+We want a way to trace, analyse and debug our distributed systems across our cloud architecture.
+
+This will give us the ability to
+
+* Find bottle necks in our services
+* Discover hidden errors within our stack
+* Identify areas of improvement
+* Justify continuous improvement work
+* Helps in diagnosis during incidents and tracking change from releases
+
+## Technical
+
+### Developer Knowledge - How well known is this in our current skill sets
+
+**Overall**: 5/10
+Our teams are starting to use this functionality in their services.
+
+Capturing data is the first part of this, however there are more things that need to be learned and implemented. We should help and encourage those wanting to learn to do so.
+
+A few areas where our maturity can be improved are.
+
+* Thinking about Observability in ticket creation as part of the Product Lifecycle
+* An understanding of what to track and what not to track
+* Teams able to continuously improve bottlenecks found in their workflow
+* Teams are able to see their service health and actively check
+* Understand and set their own benchmarks for what good looks like
+* Implement the right level of alerting to support observability
+* Look at the possibility of a Observability Community of Practice to help share learnings across teams
+
+### Decision
+
+We should use [AWS X-Ray](https://aws.amazon.com/xray/) to monitor and maintain a healthy service.
```

Comment:
Thinking about this. Is the actual decision that we should build in tracing to our products from the start and that XRay is our current preferred tool?

There's no reason why if we built it with a wrapper we couldn't use a different tool if Xray was lacking.

Reply:
So I have been wrestling with this thought, is this ADR saying we will use xray because we are already using AWS and therefore we are making this decision to use it, or should we say we will use tracing and that just means when we come to build we use xray or another solution.

I may change this to be tracing and then moving forward not mention any AWS Services, just the technology underpinning it.